### PR TITLE
Solution to issue #1056

### LINKED ISSUE #1056: Login Issue
**Actual Behaviour**

On login it throws error of unknown host when the internet is not connected

**Expected Behaviour**

It should throws some toast msg and request the user to turn on the internet.

**Steps to reproduce it**
 Try to login without internet 

**LogCat for the issue**

11-26 02:49:51.410 29192-30198/ai.susi D/OkHttp: <-- HTTP FAILED: java.net.UnknownHostException: Unable to resolve host "api.susi.ai": No address associated with hostname
11-26 02:49:51.441 29192-29192/ai.susi W/System.err: java.net.UnknownHostException: Unable to resolve host "api.susi.ai": No address associated with hostname
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at java.net.InetAddress.lookupHostByName(InetAddress.java:457)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at java.net.InetAddress.getAllByNameImpl(InetAddress.java:252)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at java.net.InetAddress.getAllByName(InetAddress.java:215)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.Dns$1.lookup(Dns.java:39)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:173)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.RouteSelector.nextProxy(RouteSelector.java:139)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.RouteSelector.next(RouteSelector.java:81)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.StreamAllocation.findConnection(StreamAllocation.java:172)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:123)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.StreamAllocation.newStream(StreamAllocation.java:93)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.HttpEngine.connect(HttpEngine.java:296)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.internal.http.HttpEngine.sendRequest(HttpEngine.java:248)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.RealCall.getResponse(RealCall.java:243)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:201)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:212)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:190)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at org.fossasia.susi.ai.rest.interceptors.TokenInterceptor.intercept(TokenInterceptor.java:25)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:190)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:163)
11-26 02:49:51.442 29192-29192/ai.susi W/System.err:     at okhttp3.RealCall.access$100(RealCall.java:30)
11-26 02:49:51.443 29192-29192/ai.susi W/System.err:     at okhttp3.RealCall$AsyncCall.execute(RealCall.java:127)
11-26 02:49:51.443 29192-29192/ai.susi W/System.err:     at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
11-26 02:49:51.443 29192-29192/ai.susi W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
11-26 02:49:51.443 29192-29192/ai.susi W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
11-26 02:49:51.443 29192-29192/ai.susi W/System.err:     at java.lang.Thread.run(Thread.java:818)
11-26 02:49:51.443 29192-29192/ai.susi W/System.err: Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
11-26 02:49:51.443 29192-29192/ai.susi W/System.err:     at libcore.io.Posix.android_getaddrinfo(Native Method)
11-26 02:49:51.445 29192-29192/ai.susi W/System.err:     at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:55)
11-26 02:49:51.445 29192-29192/ai.susi W/System.err:     at java.net.InetAddress.lookupHostByName(InetAddress.java:438)
11-26 02:49:51.445 29192-29192/ai.susi W/System.err: 	... 24 more

**Would you like to work on the issue?**

Yes! @chiragw15

### PR DESCRIPTION
**Pull request by Ananthusubramanian**
Fixes #1056
Changes: Added NetworkUtil class for help
Screenshots for the change:

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt [hunk 1]
```diff
@@ -103,12 +103,11 @@ class LoginPresenter(loginActivity: LoginActivity): ILoginPresenter, ILoginModel
...
         loginView?.showProgress(false)
 
-        if (throwable is UnknownHostException) {
-            loginView?.onLoginError(utilModel.getString(R.string.unknown_host_exception), throwable.message.toString())
-        } else {
-            loginView?.onLoginError(utilModel.getString(R.string.error_internet_connectivity),
-                    utilModel.getString(R.string.no_internet_connection))
-        }
+       if (throwable is UnknownHostException) {
+            if(NetworkUtils.isNetworkConnected()){loginView?.onLoginError(utilModel.getString(R.string.unknown_host_exception), throwable.message.toString())}
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt]: Refract this please. Use this style :
```
If () {
} else {
}
```
These small things matter.

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt [hunk 2]
```diff
@@ -81,12 +81,10 @@ class SignUpPresenter(signUpActivity: SignUpActivity) : ISignUpPresenter, ISignU
 
     override fun onError(throwable: Throwable) {
         signUpView?.showProgress(false)
-
         if (throwable is UnknownHostException) {
-            signUpView?.onSignUpError(utilModel.getString(R.string.unknown_host_exception), throwable.message.toString())
-        } else {
-            signUpView?.onSignUpError(utilModel.getString(R.string.error_internet_connectivity),
-                    utilModel.getString(R.string.no_internet_connection))
+            if(NetworkUtils.isNetworkConnected()){signUpView?.onSignUpError(utilModel.getString(R.string.unknown_host_exception), throwable.message.toString())}
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: here also

File: app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt (hunk 1 above)
Comment (Ananthusubramanian) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt]: Sorry for the mistake ,I will clear that error soon

**Comment by Ananthusubramanian** [comment]
@chiragw15 I have updated the branch

**Pull request merged**